### PR TITLE
Remove dependency of @autorest/csharp

### PR DESCRIPTION
**Pull request by pshao25**
Removed dependency of @autorest/csharp for @autorest/openapi-to-typespec

### CODE REVIEW

File: tools/tsp-client/src/commands.ts [hunk 1]
```diff
@@ -360,35 +360,7 @@ export async function convertCommand(argv: any): Promise<void> {
...
-    }
-    try {
-      await rename(joinPaths(outputDir, "metadata.json"), joinPaths(outputDir, "resources.json"));
-    } catch (err) {
-      Logger.error(
-        `Error occurred while attempting to rename metadata.json to resources.json: ${err}`,
-      );
-      process.exit(1);
-    }
     args.push("--isArm");
```
Comment (catalinaperalta) [inline tools/tsp-client/src/commands.ts:364]: So we wont need to generate the resources.json anymore? Folks only have to pass in the --isArm flag?

**Review by catalinaperalta — APPROVED** [review]

File: tools/tsp-client/package-lock.json [hunk 2]
```diff
@@ -9,8 +9,7 @@
       "version": "0.12.2",
```
Comment (catalinaperalta) [inline tools/tsp-client/package-lock.json:9]: Actually this looks wrong like the package-lock.json wasnt updated correctly. If you do an npm install is should fix all dependencies based on package.json.

**Review by catalinaperalta — CHANGES_REQUESTED** [review]
Please update package-lock.json before merging.

File: tools/tsp-client/src/commands.ts (hunk 1 above)
Comment (pshao25) [reply tools/tsp-client/src/commands.ts:364]: We generate resources.json in the openapi-to-typespec itself. So we don't need to call `@autorest/csharp`

File: tools/tsp-client/CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,10 @@
 # Release
 
+## 2024-09-09 - 0.13.0
```
Comment (catalinaperalta) [inline tools/tsp-client/CHANGELOG.md:3]:
```suggestion
## 2024-09-10 - 0.13.0
```

**Review by catalinaperalta — APPROVED** [review]

**Pull request merged**